File: web/src/api/system-user.ts
import { http } from "@/utils/http";
import type { ApiResponse, PagedData, PagedRequest } from "./customer";

export interface SystemUser {
  id: number;
  companyId: number;
  username: string;
  nickname: string;
  avatar: string;
  roles: string[];
  permissions: string[];
  isActive: boolean;
  permissionVersion: number;
  orgUnits: SystemUserOrgUnit[];
  createdAt: string;
  updatedAt?: string | null;
}

export interface SystemUserOrgUnit {
  orgUnitId: number;
  orgUnitName: string;
  orgUnitType: number;
  isPrimary: boolean;
  startAt?: string | null;
  endAt?: string | null;
}

export interface SystemUserListRequest extends PagedRequest {
  isActive?: boolean;
}

export interface CreateSystemUserRequest {
  username: string;
  password: string;
  nickname: string;
  avatar?: string | null;
  roles: string[];
  primaryOrgUnitId?: number | null;
  orgUnitIds?: number[];
  roleStartAt?: string | null;
  roleEndAt?: string | null;
  orgStartAt?: string | null;
  orgEndAt?: string | null;
  isActive: boolean;
}

export interface UpdateSystemUserRequest {
  nickname: string;
  avatar?: string | null;
  roles: string[];
  primaryOrgUnitId?: number | null;
  orgUnitIds?: number[];
  roleStartAt?: string | null;
  roleEndAt?: string | null;
  orgStartAt?: string | null;
  orgEndAt?: string | null;
  isActive: boolean;
}

export interface UpdateSystemUserStatusRequest {
  isActive: boolean;
}

export interface ResetSystemUserPasswordRequest {
  newPassword: string;
}

const baseUrl = "/api/system-users";

export const getSystemUserList = (params?: SystemUserListRequest) => {
  return http.request<ApiResponse<PagedData<SystemUser>>>("get", baseUrl, {
    params
  });
};

export const getSystemUserById = (id: number) => {
  return http.request<ApiResponse<SystemUser>>("get", `${baseUrl}/${id}`);
};

export const createSystemUser = (data: CreateSystemUserRequest) => {
  return http.request<ApiResponse<SystemUser>>("post", baseUrl, { data });
};

export const updateSystemUser = (id: number, data: UpdateSystemUserRequest) => {
  return http.request<ApiResponse<SystemUser>>("put", `${baseUrl}/${id}`, {
    data
  });
};

export const updateSystemUserStatus = (
  id: number,
  data: UpdateSystemUserStatusRequest
) => {
  return http.request<ApiResponse<SystemUser>>(
    "put",
    `${baseUrl}/${id}/status`,
    { data }
  );
};

export const resetSystemUserPassword = (
  id: number,
  data: ResetSystemUserPasswordRequest
) => {
  return http.request<ApiResponse<void>>("put", `${baseUrl}/${id}/password`, {
    data
  });
};

export const deleteSystemUser = (id: number) => {
  return http.request<ApiResponse<void>>("delete", `${baseUrl}/${id}`);
};

File: src/AcceptanceSpecSystem.Data/Entities/SystemUser.cs
namespace AcceptanceSpecSystem.Data.Entities;

/// <summary>
/// 系统用户实体（用于登录鉴权）
/// </summary>
public class SystemUser
{
    /// <summary>
    /// 主键ID
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// 所属公司ID（单公司边界）
    /// </summary>
    public int CompanyId { get; set; }

    /// <summary>
    /// 用户名（唯一）
    /// </summary>
    public string Username { get; set; } = string.Empty;

    /// <summary>
    /// 密码哈希（PBKDF2）
    /// </summary>
    public string PasswordHash { get; set; } = string.Empty;

    /// <summary>
    /// 昵称
    /// </summary>
    public string Nickname { get; set; } = string.Empty;

    /// <summary>
    /// 头像地址
    /// </summary>
    public string Avatar { get; set; } = string.Empty;

    /// <summary>
    /// 是否启用
    /// </summary>
    public bool IsActive { get; set; } = true;

    /// <summary>
    /// 权限版本号（角色或组织变更后递增，可用于强制令牌刷新）
    /// </summary>
    public int PermissionVersion { get; set; } = 1;

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    /// <summary>
    /// 更新时间
    /// </summary>
    public DateTime? UpdatedAt { get; set; }

    /// <summary>
    /// 导航属性：所属公司
    /// </summary>
    public OrgCompany Company { get; set; } = null!;

    /// <summary>
    /// 导航属性：用户角色关系
    /// </summary>
    public ICollection<AuthUserRole> UserRoles { get; set; } = new List<AuthUserRole>();

    /// <summary>
    /// 导航属性：用户组织关系
    /// </summary>
    public ICollection<AuthUserOrgUnit> UserOrgUnits { get; set; } = new List<AuthUserOrgUnit>();
}
